### ISSUE TITLE
Implement new reporting form

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons-frontend#12432]

**Comment by ioanarusiczki** [comment]
@eviljeff 

I don't see a change on the Abuse Report form on -dev right now. Do I need to set a switch in admin to test it? 

I tested the current form on -stage at release - looks good.

**Comment by eviljeff** [comment]
sorry, forgot to include more details.  The current abuse reporting component hasn't been switched out yet (#12485)
the form is now at a separate url https://addons-dev.allizom.org/en-US/firefox/feedback/addon/<slug or guid or numeric id>/

**Comment by eviljeff** [comment]
re: stage (and prod) - the form is behind a feature flag so should only be accessible on dev for now.

**Comment by ioanarusiczki** [comment]
@eviljeff 

Yes, the urls on -stage or -prod throw a 404, they're not accessible.  

- I could see the new form using a slug/guid for any add-on type -> used a ids of extensions, themes, dict or lang pack 
An example: https://addons-dev.allizom.org/en-US/firefox/feedback/addon/uBlock0@raymondhill.net/

- there are 6 reasons (the current form has more than 6, the admin even more than the current form ) only one can be checked. I think all this is expected for now. 
- submit report can be sent only with a reason, details and checking " I hereby certify,...." (unchecking it the attempt to submit would throw a warning 

![warning](https://github.com/mozilla/addons-frontend/assets/33448286/01ca304d-d18d-417c-ac20-165c6316da19)

- I saw the reports present in admin in the abuse report model. 

![new abuse report form](https://github.com/mozilla/addons-frontend/assets/33448286/e4090fbb-5cdc-4212-a7f6-24901b6abbeb)

**Comment by ioanarusiczki** [comment]
@eviljeff I forgot to ask, is this going to be implemented on Android too?

**Comment by eviljeff** [comment]
> @eviljeff I forgot to ask, is this going to be implemented on Android too?

yes - it's the same form though (with `/android/` rather than `/firefox/`)

**Comment by ioanarusiczki** [comment]
@eviljeff 

I've checked : 
- Location field , which wasn't present yesterday , sent with all options (or none) -> Location field in admin receives what's been sent through the form.  
- Location is optional, mandatory fields are: reason, details and checking the agreement below the detail field.
- with login: the Full Name field is auto-completed with the users's display name,  the e-mail adress is also auto-filled
- logout view: the name and e-mail fields are optional, report can be sent without input from the user, e-mail field required a specific e-mail format.
- I filed an issue on Android https://github.com/mozilla/addons-frontend/issues/12508.

**Comment by KevinMind** [comment]
Old Jira Ticket: https://mozilla-hub.atlassian.net/browse/ADDFRNT-234
